tutorial/cdn-caching.md

```diff
@@ -0,0 +1,158 @@
+
+4. Format for collections:
+
+   ```javscript
+   `{c/${publisherId}/${collection.id}}`
+   ```
+
+   Eg:
+
+   `< cache-tag: c/1166/90576, s/1166/5b10a46a, s/1166/185a3cb0`
+
+   The front end gets updated after it sees the cache keys from the bold when the collection gets updated and published.
+
+5. Format for story:
+
+   ```javscript
+   `{s/${publisherId}/${story.id.substr(0, 8)}}`
+   ```
+
+   Eg:
+
+   `< cache-tag: s/1166/01166259, a/1166/930893`
+
+   The story page or the particluar story in listing/collection page with the changed content gets updated after the front end sees the cache-keys from the bold after the story gets updated and published.
+
+6. Format for author :
+
+   ```javscript
+   `{a/${publisherId}/${author.id}}`
+   ```
+
+   The author cache key gets updated when author details are changed/updated.
+
+How Quintype sets the cache headers ?
+
+- If cache keys are present
+
+  1. If the CDN provider is `Cloudflare`/`Fastly` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Cache-Control" | `public,max-age=15,s-maxage=${sMaxAge},stale-while-revalidate=1000,stale-if-error=14400`
+        "Vary" | "Accept-Encoding"
+        "Cache-Tag"  | _(cacheKeys).uniq().join(",")
+        "Surrogate-Key" | _(cacheKeys).uniq().join(" ")
+
+  2. If the CDN provider is `Akamai` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Edge-Control" | `public,maxage=${sMaxAge},stale-while-revalidate=1000,stale-if-error=14400`
+        "Vary" | "Accept-Encoding"
+        "Cache-Tag"  | _(cacheKeys).uniq().join(",")
+        "Surrogate-Key" | _(cacheKeys).uniq().join(" ")
+
+- If the value of cache keys is set to "DO_NOT_CACHE"
+
+  1. If the CDN provider is `Cloudflare`/`Fastly` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Cache-Control" | "private,no-cache,no-store,max-age=0"
+        "Vary" | "Accept-Encoding"
+
+  2. If the CDN provider is `Akamai` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Edge-Control" | "private,no-cache,no-store,max-age=0"
+        "Vary" | "Accept-Encoding"
+
+- If cache keys are not present
+
+  1. If the CDN provider is `Cloudflare`/`Fastly` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Cache-Control" | "public,max-age=15,s-maxage=60,stale-while-revalidate=150,stale-if-error=3600"
+        "Vary" | "Accept-Encoding"
+
+  2. If the CDN provider is `Akamai` then the headers are set as below:
+
+        header | value
+        --- | ---
+        "Edge-Control" | "public,maxage=60,stale-while-revalidate=150,stale-if-error=3600"
+        "Vary" | "Accept-Encoding"
+
+How Quintype handles toggle between different CDN providers ?
+
+- For now, we have support for `Cloudflare`, `Fastly`, and `Akamai`.
+- The default CDN provider is set as `Cloudflare`.
```

Comment:
Mention Fastly is used as backup CDN if Cloudflare goes down

Reply:
sure